### PR TITLE
(fix) svelte2tsx handling increment postfix, prefix and Assignment operators

### PR DESCRIPTION
**Pull request by sw-yx**
closes https://github.com/sveltejs/language-tools/issues/170

this was a fun one to solve :) `svelte2tsx.ts` has some regrettable duplication between `processInstanceScriptContent` and `processSvelteTemplate` which use `ts.createSourceFile` and `convertHtmlxToJsx` separately, hence every new language feature supported must be done for 2 different ASTs. its fine now but could grow to unmanageable scale. i am wondering whether or not it makes sense to significantly reduce the scope of `convertHtmlxToJsx` and use ts' AST internally - but havent really investigated enough to know for sure

### CODE REVIEW

File: packages/svelte2tsx/src/svelte2tsx.ts [hunk 1]
```diff
@@ -83,6 +83,41 @@ function processSvelteTemplate(str: MagicString): TemplateProcessResult {
...
+        // handle $store++, $store--, ++$store, --$store
+        // https://github.com/sveltejs/language-tools/issues/170
+        if (parent.type == 'UpdateExpression') {
+            let appendor;
+            if (parent.operator === '++') appendor = '+';
+            if (parent.operator === '--') appendor = '-';
+            if (appendor === undefined)
+                throw new Error(
+                    'warning - unrecognized UpdateExpression operator! Probably some edge case unaccounted for in svelte2tsx, please file an issue.',
+                );
```
Comment (sw-yx) [inline packages/svelte2tsx/src/svelte2tsx.ts:112]: not actually sure if this is needed but just being cautious - are there any other operators like ++ and --? i dont know.

File: packages/svelte2tsx/src/svelte2tsx.ts [hunk 2]
```diff
@@ -83,6 +83,40 @@ function processSvelteTemplate(str: MagicString): TemplateProcessResult {
             str.appendLeft(parent.end, ')');
             return;
         }
+        // handle Assignment operators ($store +=, -=, *=, /=, %=, **=)
+        const operators = ['+=', '-=', '*=', '/=', '%=', '**='];
```
Comment (Th0rN13) [inline packages/svelte2tsx/src/svelte2tsx.ts:87]: What about bitwise operators?
From [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Expressions_and_Operators)
`<<= >>= >>>= &= ^= |=`

Comment (sw-yx) [reply packages/svelte2tsx/src/svelte2tsx.ts:87]: haha. honestly doubt anyone uses bitwise operators with stores...

**Comment by halfnelson** [comment]
Nice fix. Thanks!

File: packages/svelte2tsx/src/svelte2tsx.ts (hunk 2 above)
Comment (sw-yx) [reply packages/svelte2tsx/src/svelte2tsx.ts:87]: image
dammit look what u made me do

**Comment by dummdidumm** [comment]
Nice one!

About the duplicated code: Maybe we can find a intermediate representation which both AST and ts map to, and then the logic is executed only once from this intermediate representation.

**Pull request merged**